app/serializers/npq_participant_serializer.rb

```diff
@@ -6,18 +6,30 @@ class NPQParticipantSerializer
   include JSONAPI::Serializer
   include JSONAPI::Serializer::Instrumentation
 
-  set_id :id
+  set_id :id, &:user_id
   set_type :'npq-participant'
 
-  attributes :participant_id, :email, :full_name
+  attribute(:participant_id, &:user_id)
 
-  attribute(:participant_id, &:id)
+  attribute(:npq_courses) do |npq_participant_profile|
+    npq_participant_profile.user.npq_profiles.map do |npq_profile|
+      npq_profile.npq_course.identifier
+    end
+  end
+
+  attribute(:email) do |npq_participant_profile|
+    npq_participant_profile.user.email
+  end
+
+  attribute(:full_name) do |npq_participant_profile|
+    npq_participant_profile.user.full_name
+  end
 
-  attribute(:npq_courses) do |object|
-    object.npq_profiles.map { |npq_profile| npq_profile.npq_course.identifier }
+  attribute(:teacher_reference_number) do |npq_participant_profile|
+    npq_participant_profile.user.teacher_profile&.trn
   end
 
-  attribute(:teacher_reference_number) do |object|
-    object.teacher_profile&.trn
+  attribute(:training_status) do |npq_participant_profile|
+    npq_participant_profile.profile&.state
```

Comment:
I think since you wrote it we might have changed the field to be called `training_status` to cache the status rather than to rely on loading an ordered has_many relation (cause it was buggy).

Reply:
Also, what is `.profile` in here referring to? Is `npq_participant_profile` actually `npq_validation_data`?